### ISSUE TITLE
Remote Desktop Protocol (RDP) with Botan

### ISSUE THREAD

**Issue by ahura24** [randombit/botan#2248]
Hi and thanks for your good library

We want to use Botan as the server side or the client side on RDP protocol. We configure RDP only runs in TCP mode on the port 3389 without any UDP packets. The SSL handshake is successful, but after 2 packets we get fatal error with code 80 (internal error). 

Does Botan work with RDP ?

**Comment by randombit** [comment]
So you are saying the entire handshake completes, and it is possible to exchange data, but then after a few application data packets carrying the RDP payloads, you get a TLS-level alert? It seems odd to me that RDP would use TLS alerts to indicate RDP level protocol issues...

Do you see this on both client and server side? What ciphersuite is negotiated, when you are seeing these errors? Can you point me to any documentation about the RDP protocol that might be helpful?

**Comment by ahura24** [comment]
Yes, it is. Handshake is completed with TLS_RSA_WITH_AES_256_GCM_SH384 (0X009D). It's the same time when we don't use Botan. When we use Botan in each side (client/server), we get internal alert. unfortunately I don't see any documents about RDP. It's important for us to know Botan can support RDP or not. I can send you wireshark export packets if i have your Email address.
